File: src/nodeStringBuffer.ml
type t (* can be either string or buffer *)

type _ stringBufferKind =
  | String :  Js.String.t stringBufferKind
  | Buffer :  NodeBuffer.t stringBufferKind

(** We except a good inliner will eliminate such boxing in the future *)
let test (type a) (x : t) : (a stringBufferKind * a)=
  if Js.typeof x = "string" then
    (Obj.magic String : a stringBufferKind), (Obj.magic x : a)
  else
    (Obj.magic Buffer : a stringBufferKind), (Obj.magic x : a)

File: lib/js/src/nodeStringBuffer.js
'use strict';


function test(x) {
  if (typeof x === "string") {
    return /* tuple */[
            /* String */0,
            x
          ];
  } else {
    return /* tuple */[
            /* Buffer */1,
            x
          ];
  }
}

exports.test = test;
/* No side effect */
